rbd-target-api.py

```diff
@@ -992,11 +992,11 @@ def disk(pool, image):
     **RESTRICTED**
     Examples:
     curl --insecure --user admin:admin -d mode=create -d size=1g -d pool=rbd -d count=5
-        -X PUT https://192.168.122.69:5000/api/disk/rbd.new2_
-    curl --insecure --user admin:admin -d mode=create -d size=10g -d pool=rbd
-        -X PUT https://192.168.122.69:5000/api/disk/rbd.new3_
-    curl --insecure --user admin:admin -X GET https://192.168.122.69:5000/api/disk/rbd.new2_1
-    curl --insecure --user admin:admin -X DELETE https://192.168.122.69:5000/api/disk/rbd.new2_1
+        -X PUT https://192.168.122.69:5000/api/disk/rbd/new0
+    curl --insecure --user admin:admin -d mode=create -d size=10g -d pool=rbd -d create_image=true
```

Comment:
Nit: `create_image` is implied if `size` is provided -- a better example would be to drop `size` and set `create_image=false`

Reply:
It seems not to the current code:

```
# curl --insecure --user admin:admin -d mode=create -d size=100m -d pool=rbd2 -d image=block1 -d count=1 -X PUT http://192.168.195.172:5000/api/disk/rbd2/block1
{"message":"Image rbd2/block1 does not exist"}

```

Reply:
The code:

`if mode == 'create' and (not create_image or not size):`

Nit: the `if` branch won't create the image. That means only when `size` is specified and the `create_image` must be `true` will it create the image.

Reply:
Thx -- but does that mean the previous example is also wrong since it doesn't have `create_image=true` (given the image name of `new...` implies it should be created not reused).

Reply:
The logic changed a lot, I have test the REST APIs one by one 3 years ago and the example was correct then. I will try to check and fix them if there has issues.

Reply:
Ack -- so it might be easier to leave this PR open as a running collection of fixes as you go through developing the as-built tests.

Reply:
Sure.